components/dashboard/src/components/SelectWorkspaceClassComponent.tsx

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { SupportedWorkspaceClass } from "@gitpod/gitpod-protocol/lib/workspace-class";
+import { useEffect, useMemo, useState } from "react";
+import { getGitpodService } from "../service/service";
+import WorkspaceClass from "../icons/WorkspaceClass.svg";
+import { DropDown2, DropDown2Element } from "./DropDown2";
+
+interface SelectWorkspaceClassProps {
+    selectedWorkspaceClass?: string;
+    onSelectionChange: (workspaceClass: string) => void;
+}
+
+export default function SelectWorkspaceClassComponent(props: SelectWorkspaceClassProps) {
+    const [workspaceClasses, setWorkspaceClasses] = useState<SupportedWorkspaceClass[]>();
+    useEffect(() => {
+        getGitpodService().server.getSupportedWorkspaceClasses().then(setWorkspaceClasses);
+    }, []);
+    const getElements = useMemo(() => {
+        return (search: string) => {
+            if (!workspaceClasses) {
+                return [];
+            }
+            return [
+                ...workspaceClasses
+                    .filter(
+                        (c) =>
+                            `${c.description}${c.displayName}${c.category}`
+                                .toLowerCase()
+                                .indexOf(search.toLowerCase()) !== -1,
+                    )
+                    .map(
+                        (c) =>
+                            ({
+                                id: c.id,
+                                element: <WorkspaceClassDropDownElement wsClass={c} />,
+                                isSelectable: true,
+                            } as DropDown2Element),
+                    ),
+            ];
+        };
+    }, [workspaceClasses]);
+    return (
+        <DropDown2
+            getElements={getElements}
+            onSelectionChange={props.onSelectionChange}
+            searchPlaceholder="Select class"
```

Comment:
**suggestion**: Not sure if this is the best line to comment, but let's remove the select class hint above the class options.

Reply:
With no replacement?

Reply:
I've just removed it now.